File: cod-code/src/com/mycompany/entities/Menu.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package com.mycompany.entities;
/**
 *
 * @author RoKy-Dev
 */
public class Menu {
    private int id,price;
    private String name,category,description;

    public Menu() {
    }

    public Menu( int price, String name, String category, String description) {
        this.price = price;
        this.name = name;
        this.category = category;
        this.description = description;
    }

    @Override
    public String toString() {
        return "Menu{" + "id=" + id + ", price=" + price + ", name=" + name + ", category=" + category + ", description=" + description + "\n";
    }
 public String idtoString ()
    {
        return String.valueOf(id);
    }
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getPrice() {
        return price;
    }

    public void setPrice(int price) {
        this.price = price;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getCategory() {
        return category;
    }

    public void setCategory(String category) {
        this.category = category;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

}

File: cod-symfony/src/Entity/Menu.php
<?php

namespace App\Entity;

use App\Repository\MenuRepository;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;
use Symfony\Component\Serializer\Annotation\Groups;

/**
 * @ORM\Entity(repositoryClass=MenuRepository::class)
 */
class Menu
{
    /**
     * @ORM\Id
     * @ORM\GeneratedValue
     * @ORM\Column(type="integer")
     * @Groups("Menu")
     * @Groups("Inorder")
     */
    private $id;

    /**
     * @Assert\NotBlank(message="field can't be empty")
     * @Assert\Length(
     *      min = 5,
     *      minMessage="The minimum length is 5"
     *
     *     )
     * @ORM\Column(type="string", length=255)
     * @Groups("Menu")
     */
    private $name;

    /**
     *  @Assert\NotBlank(message="field can't be empty")
     *   @Assert\Range(
     *      min = 1,
     *      max = 500,
     *      minMessage = "Minimum Price Is 1",
     *      maxMessage = "Maximum Price is 500"
     * )
     * @ORM\Column(type="integer")
     * @Groups("Menu")
     */
    private $price;

    /**
     * @Assert\NotBlank(message="field can't be empty")
     * @Assert\Length(
     *      min = 5,
     *      minMessage="The minimum length is 5"
     *
     *     )
     * @ORM\Column(type="string", length=255)
     * @Groups("Menu")
     */
    private $category;

    /**
     * @Assert\NotBlank(message="field can't be empty")
     * @Assert\Length(
     *      min = 7,
     *      max = 100,
     *      minMessage = "The minimum length is 7 ",
     *      maxMessage = "The maximum length is 100" )
     * @ORM\Column(type="string", length=255)
     * @Groups("Menu")
     */
    private $description;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getName(): ?string
    {
        return $this->name;
    }

    public function setName(string $name): self
    {
        $this->name = $name;

        return $this;
    }

    public function getPrice(): ?int
    {
        return $this->price;
    }

    public function setPrice(int $price): self
    {
        $this->price = $price;

        return $this;
    }

    public function getCategory(): ?string
    {
        return $this->category;
    }

    public function setCategory(string $category): self
    {
        $this->category = $category;

        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(string $description): self
    {
        $this->description = $description;

        return $this;
    }
    public function __toString()
    {
        return(string)$this->getName();
    }
}
